### DISCUSSION TITLE
Any way to scale the ui?

### DISCUSSION THREAD

**Discussion by Cheakerdoodels** [iina/iina#4204] [Q&A]
Is there any way to scale the UI? I find that using iina from my couch I can barely make out things like chapter markers on the progress bar or the time remaining in a video? You can make it somewhat bigger by changing the entire os ui render scale but that seems excessive and I'd like to be able to just change the apps.

**Comment by low-batt** [comment]
I do not believe it is possible to scale up the On Screen Controller. The OSC is the UI component that contains the progress bar, time labels and chapter markers (if `Show chapter position in progress bar` is enabled).

It is possible to scale up the On Screen Display. The OSD is the UI component that appears in the upper left side of the main window. This component reports status. For example it will appear when playback is paused and resumed. To increase the size of the OSD follow these instructions:
- Start IINA
- Click on  `Settings…` under the `IINA` menu
- The settings panel appears
- On the left side of the panel click on `UI`
- On the right side of the panel scroll down to the `On Screen Display` section
- Change the `Text size` setting as desired

This assumes you have not enabled mpv's OSD in place of IINA OSD which is possible in `Advanced` settings.

The difficulty of seeing chapter markers has been reported in issues #2544 and #3860.

**Reply by Cheakerdoodels** [reply]
The OSD font scale helps mitigate my issue quite a lot but I don't seem to be able to set a hotkey to make the osd show up without having to just pause and unpause. The standard mpv show-progress command doesn't seem to work when set to a keybind. Is the command different for iina?

**Comment by low-batt** [comment]
I'm not aware of a way to bind a key such that it triggers the IINA OSD.

To make use of [show-progress](https://mpv.io/manual/stable/#command-interface-show-progress) I think you will have to switch to the mpv OSD. The following instructions configure IINA to use the mpv OSD and configures the behavior of the mpv OSD using [osd-bar](https://mpv.io/manual/stable/#options-osd-bar), [osd-font-size](https://mpv.io/manual/stable/#options-osd-font-size), and [osd-status-msg](https://mpv.io/manual/stable/#options-osd-status-msg):
- Start IINA
- Click on  `Settings…` under the `IINA` menu
- The settings panel appears
- On the left side of the panel click on `Advanced`
- Slide the `Enable advanced settings` toggle button to be on (blue)
- Check the `Use mpv's OSD` setting
- In the `Additional mpv options` section click on `+`
- A new entry appears in the table
- Double click on `name` and replace it with `osd-bar`
- Double click on `value` and replace it with `no`
- Click on  `+`, set `name` to `osd-font-size`, `value` to `100`
- Click on  `+`, set `name` to `osd-status-msg`, `value` to `${time-pos}`
- Confirm the settings panel looks like the screenshot below
- **Restart IINA** to activate the new settings
![discussion-4204](https://user-images.githubusercontent.com/86170219/216520605-a395466f-5cd0-4d66-95c6-3a01d1ebc108.jpg)

The key you have bound to `show-progress` should now display the mpv OSD in the upper left corner of the main window. Enabling the mpv OSD disable's IINA's OSD, so only messages normally shown by mpv will be displayed. See the [OSD](https://mpv.io/manual/stable/#osd) section of the mpv manual for details on configuring mpv's OSD.
